api/routers/edr.py

```diff
@@ -73,7 +77,11 @@ async def get_data_location_id(
     get_obs_request = dstore.GetObsRequest(
         filter=dict(
             platform=dstore.Strings(values=[location_id]),
-            instrument=dstore.Strings(values=list(map(str.strip, parameter_name.split(",")))),
+            **(
+                {"instrument": dstore.Strings(values=list(map(str.strip, parameter_name.split(","))))}
+                if parameter_name
+                else {}
+            ),
```

Comment:
How about something like this (untested):
```
    filter = dict(platform=dstore.Strings(values=[location_id]))
    if parameter_name:
        dict["instrument"] = dstore.Strings(values=list(map(str.strip, parameter_name.split(","))))
    get_obs_request = dstore.GetObsRequest(
        filter=filter,
        temporal_interval=dstore.TimeInterval(start=range[0], end=range[1]) if range else None,
    )
```

Reply:
Has the same effect and I guess that would be easier to read & add new filters. I guess I'll do:
```
    filter = {}
    if parameter_name: 
        filter=dict(instrument=dstore.Strings(values=list(map(str.strip, parameter_name.split(",")))))

    range = get_datetime_range(datetime)
    get_obs_request = dstore.GetObsRequest(
        filter=filter,
        spatial_area=dstore.Polygon(
            points=[dstore.Point(lat=coord[1], lon=coord[0]) for coord in poly.exterior.coords]
        ),
        temporal_interval=dstore.TimeInterval(start=range[0], end=range[1]) if range else None,
    )
```
with the area endpoint then.

Reply:
https://github.com/EURODEO/e-soh/pull/30/commits/f6787791d2654b40dbe37798c10f0fd85358e49c